### ISSUE TITLE
Enable Gated Builds/Steps from Pipelines

### ISSUE THREAD

**Issue by jbrockopp** [go-vela/community#55]
## Description

Enable the ability to have "gated" builds/steps

From Sussman (@JordanSussman):

> We heavily utilize https://jenkins.io/doc/pipeline/steps/pipeline-input-step/ for our infrastructure deployments (kubernetes, aws, iam, etc). The workflow looks like:
> 1. Developer commits to repo and opens pr
> 2. Jenkins runs tests against all environments (dev, stage, prod, etc.)
> 3. Code is merged into master
> 4. Rollout to each environment is gated by someone clicking the accept button
>
> Allows us to easily roll through all environments without needing to funky branching or tagging strategies in github
> Also enables us to do some manual validation that the individual environment deployment didn't cause undesired issue before proceeding to the next environment.

## Value

Allow the ability to have "push button" deploys while still allowing manual validation for the environment

## Definition of Done

A user can configure Vela to "gate" running a build/step until a manual approval is provided via button in UI or command in CLI.

## Effort  (Optional)

2 weeks

## Impacted Personas (Optional)

everyone

**Comment by plyr4** [comment]
implemented through https://github.com/go-vela/server/pull/1016
while thats for the entire build, the spirit of the issue is complete.
further enhancement ideas and issues are welcome! :D
